srtcore/buffer.cpp

```diff
@@ -63,6 +63,41 @@ using namespace std;
 using namespace srt_logging;
 using namespace srt::sync;
 
+bool AvgBufSize::canUpdate(const time_point& now) const
+{
+    const int      SRT_MAVG_BASE_PERIOD = 1000000; // us
+    const int      SRT_us2ms            = 1000;
+    const uint64_t elapsed_ms           = count_milliseconds(now - m_tsLastSamplingTime); // ms since last sampling
+    return (elapsed_ms >= (SRT_MAVG_BASE_PERIOD / SRT_MAVG_SAMPLING_RATE) / SRT_us2ms);
+}
+
+void AvgBufSize::update(const steady_clock::time_point& now, int pkts, int bytes, int timespan_ms)
+{
+    const uint64_t elapsed_ms = count_milliseconds(now - m_tsLastSamplingTime); // ms since last sampling
+
+    m_tsLastSamplingTime = now;
+
+    if (elapsed_ms > 1000)
+    {
+        // No sampling in last 1 sec, initialize average
```

Comment:
Is assumption, that now > 1000 will always be true, safe?

Reply:
It is unlikely to be true. Most of the time elapsed_ms is below 1s.

Reply:
I mean, a situation that `now` would ship a value less than 1000. Then when you subtract 0 from it you'll get the same value, and already less than 1000. If not, then the "too long time ago" situation would be handled together with "didn't initialize before".

Reply:
> `now` would ship a value less than 1000

`now` is the current time. How can it ship a value of less than 1000?
Or do you mean `now - m_tsLastSamplingTime`?

Reply:
No, I meant `now` exactly - so that the result of the value compared against 1000 would be less than 1000. I personally think that even if you have a steady clock that starts from 0 at the moment when the system starts, no system in the world is capable of running the application in less than 1s after bootup.